### PR TITLE
Correlation RFC.

### PR DESCRIPTION
**Pull request by mfelsche**
[Rendered](https://github.com/tremor-rs/tremor-rfcs/blob/correlation/text/0000-correlation.md)

# Checklist

* [x] Add a link to rendered text (see above)
* [x] Assign core group reviewer (manual: assign review)
* [x] Mark as ready for review / non-draft when ready for open review (manual: move to open for review)
* [ ] Core reviewer accepts the once the open review period is over, and there are no more outstanding issues (automation: accepted)
* [ ] Assign RFC number by the end of the review period

### CODE REVIEW

File: text/0000-correlation.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+Explain the proposal as if it was already included in Tremor and you were teaching it to another Tremor user. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how stakeholders should *think* about the feature, and how it should impact the way they use tremor. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing tremor stakeholders and new tremor programmers.
+
+For implementation-oriented RFCs (e.g. for language internals), this section should focus on how language contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
```
Comment (anupdhml) [inline text/0000-correlation.md:73]: remove

File: text/0000-correlation.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+define tumbling window size_2
+with
+  size = 2,
+  eviction_period = 1000
+end;
+
+select aggr::win::collect_flattened(event) from in[size_2] group by $correlation into out;
+```
+
+To get exactly the same timeout behaviour the operator would have, we might need to tweak the current `eviction_period` handling logic, as it currently only gets rid of groups after `2 x eviction_perios`. But as the operator can only live in 1 pipeline, we need to pass both events through the same pipeline anyways, and the above code is much more idiomatic and feels more native and less cumbersome.
```
Comment (anupdhml) [inline text/0000-correlation.md:108]: typo: `2 x eviction_periods`

**Review by anupdhml — COMMENTED** [review]
nice writeup!

**Review by darach — COMMENTED** [review]
It may be opportune to explicitly allow support for OpenTelemetry so that correlatable events can be exposed via OpenTelemetry trace instrumentation

**Comment by mfelsche** [comment]
You mean adding opentelemetry support to tremor itself, and creating trace spans for our events that we export in some way? via https://crates.io/crates/opentelemetry-otlp or something similar?

**Pull request merged**